sympy/parsing/latex/lark/transformer.py

```diff
@@ -42,10 +42,17 @@ class TransformToSymPyExpr(Transformer):
 
         Note that the option must be set to ``True`` for the default parser to work.
     """
+    def __init__(self, evaluate):
+        self.evaluate = evaluate
 
     SYMBOL = sympy.Symbol
     DIGIT = sympy.core.numbers.Integer
 
+    def transform_tree(self, tree, evaluate):
+        self.evaluate = evaluate
+
+        return super().transform(tree)
```

Comment:
I don't think that you need to modify `transform_tree`.

Reply:
The problem with this is that it disallows the user from doing something like
```python
expr1 = parse_latex(r"\frac12", evaluate=False)
expr2 = parse_latex(r"\int_1^2 \dfrac{\mathrm{d}x}{x}", evaluate=True)
```
That is, it prevents the user from evaluating in one call and not evaluating in another `parse_latex` call, or vice versa.

Reply:
Why do you have to save the Transformer object to LarkLaTeXParser persistently?

We can construct new ‘Transformer’ with according options just before parsing, than saving it persistently to the parser.

For example, SymPy printer functions already does that, it generates new Printer object every time, and we didn’t get any performance downside of doing that than other performance issues. 
I’m aware that initializing Lark is slow because it compiles the grammar, but it is very cheap to initialize Transformer